server/app/views/admin/programs/ProgramBlockPredicatesEditViewV2.java

```diff
@@ -0,0 +1,244 @@
+package views.admin.programs;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.label;
+import static j2html.TagCreator.text;
+import static play.mvc.Http.HttpVerbs.POST;
+
+import com.google.common.collect.ImmutableList;
+import controllers.admin.routes;
+import j2html.tags.specialized.DivTag;
+import j2html.tags.specialized.FormTag;
+import j2html.tags.specialized.InputTag;
+import j2html.tags.specialized.LabelTag;
+import java.util.UUID;
+import javax.inject.Inject;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import services.program.BlockDefinition;
+import services.program.ProgramDefinition;
+import services.question.types.QuestionDefinition;
+import views.HtmlBundle;
+import views.admin.AdminLayout;
+import views.admin.AdminLayout.NavPage;
+import views.admin.AdminLayoutFactory;
+import views.components.Icons;
+import views.components.LinkElement;
+import views.components.ToastMessage;
+import views.style.ReferenceClasses;
+
+/** Renders a page for editing predicates of a block in a program. */
+public final class ProgramBlockPredicatesEditViewV2 extends ProgramBlockView {
```

Comment:
note: this class is forked from ProgramBlockPredicatesEditView

Reply:
ok, and you basically took the editor control creation code out of it and moved it into the ConfigureView class correct?

Reply:
Yes